src/controllers.cpp

```diff
@@ -17,76 +17,114 @@
-#include "GameController.hpp"
 #include "SerieMainMenuController.hpp"
 #include "OfficialSeriesSelectorController.hpp"
-#include "Messages.hpp"
-
-#include <iostream>
+#include "GameController.hpp"
 
-using namespace factories;
+namespace memoris
+{
+namespace controllers
+{
 
 /**
  *
  */
-controllers::Controller* ControllerFactory::getControllerById(
-    unsigned short id,
+Controller* getControllerById(
+    const unsigned short& id,
     utils::Context& context
 )
 {
+    /* return a pointer to the correct controller according to the id */
     switch(id)
     {
-    case MAIN_MENU_CONTROLLER_ID:
-        return new controllers::MainMenuController;
     case NEW_GAME_CONTROLLER_ID:
-        return new controllers::NewGameController;
+    {
+        return new NewGameController;
+    }
+    break;
     case EDITOR_MENU_CONTROLLER_ID:
-        return new controllers::EditorMenuController;
+    {
+        return new EditorMenuController;
+    }
+    break;
     case EDITOR_SERIE_CONTROLLER_ID:
-        return new controllers::EditorSerieController(context);
+    {
+        return new EditorSerieController(context);
+    }
+    break;
     case OPEN_SERIE_CONTROLLER_ID:
-        return new controllers::OpenSerieController;
+    {
+        return new OpenSerieController;
+    }
+    break;
     case EDITOR_LEVEL_CONTROLLER_ID:
-        return new controllers::EditorLevelController;
+    {
+        return new EditorLevelController;
+    }
+    break;
     case OPEN_LEVEL_CONTROLLER_ID:
-        return new controllers::OpenLevelController;
+    {
+        return new OpenLevelController;
+    }
+    break;
+    case SERIE_MAIN_MENU_CONTROLLER_ID:
+    {
+        return new SerieMainMenuController;
+    }
+    break;
+    case OFFICIAL_SERIES_SELECTOR_CONTROLLER_ID:
+    {
+        return new OfficialSeriesSelectorController;
+    }
+    break;
     case GAME_CONTROLLER_ID:
+    {
+        /* the game controller constructor tries to open the level file;
+           the process to open the file is handled by the FileWriter; if the
+           read process fails, an exception is thrown and the error controller
+           with a specific message for this case is rendered instead; the
+           error message is also set in the FileWriter class */
         try
         {
-            return new controllers::GameController(
-                       context.getNxtLvlStrPath()
-                   );
+            return new GameController(context.getNxtLvlStrPath());
         }
         catch(const std::invalid_argument& e)
         {
             return getErrCtrl(e.what());
         }
-    case SERIE_MAIN_MENU_CONTROLLER_ID:
-        return new controllers::SerieMainMenuController;
-    case OFFICIAL_SERIES_SELECTOR_CONTROLLER_ID:
-        return new controllers::OfficialSeriesSelectorController;
+    }
+    break;
+
+    /* by default, if the controller id does not exist, the main
+       menu is rendered; it avoids mistakes in screens transitions */
```

Comment:
add that "maybe", the music won't be correct, but everything is solved when the controller is switched or the program terminated...